### PR TITLE
[SPARK-11450][SQL] Add Unsafe Row processing to Expand

### PR DESCRIPTION
**Pull request by hvanhovell**
This PR enables the Expand operator to process and produce Unsafe Rows.

### CODE REVIEW

**Comment by marmbrus** [comment]
ok to test

File: sql/core/src/main/scala/org/apache/spark/sql/execution/Expand.scala [hunk 1]
```diff
@@ -41,14 +41,34 @@ case class Expand(
...
 
+  override def outputsUnsafeRows: Boolean = child.outputsUnsafeRows
+  override def canProcessUnsafeRows: Boolean = true
+  override def canProcessSafeRows: Boolean = true
+
+  protected val cleanedProjections = {
+    if (outputsUnsafeRows) {
+      projections.map { exprs =>
+        exprs.map(_ transform {
+          case CreateStruct(children) => CreateStructUnsafe(children)
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/Expand.scala]: Should we do this inside of `UnsafeProjection.create`?  It seems weird to have to remember to do it in each place we use it.

Comment (hvanhovell) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/Expand.scala]: That would be much better. I'll make the change.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/ExpandSuite.scala [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    val attributes = projections.head.map(_.toAttribute)
+    checkAnswer(
+      input.toDF(),
+      plan => Expand(projections, attributes, f(plan)),
+      input.flatMap(i => Seq.tabulate(2)(j => Row(i._1, j)))
+    )
+  }
+
+  test("expanding UnsafeRows") {
+    testExpand(ConvertToSafe)
```
Comment (hvanhovell) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/ExpandSuite.scala]: ConvertToUnsafe... :(

**Comment by marmbrus** [comment]
Thanks!  Merging to master and 1.6.

**Pull request closed without merging**